src/intrinsics/ecma262/Object.js

```diff
@@ -72,8 +72,11 @@ function handleObjectAssignSnapshot(
         frm.makePartial();
         delayedSources.push(frmSnapshot);
       }
+    } else {
+      return false;
     }
   }
+  return true;
```

Comment:
Would be better for the future to add this next to all the delayedSources.push lines so that if we add another branch, it isn't just implied.